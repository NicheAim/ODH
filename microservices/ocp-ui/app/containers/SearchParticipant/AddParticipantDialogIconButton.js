import styled from 'styled-components';
import IconButton from 'material-ui/IconButton';

const AddParticipantDialogIconButton = styled(IconButton).attrs({
  style: { top: '26px', height: '30px' },
})('');

AddParticipantDialogIconButton.propTypes = {};

export default AddParticipantDialogIconButton;
