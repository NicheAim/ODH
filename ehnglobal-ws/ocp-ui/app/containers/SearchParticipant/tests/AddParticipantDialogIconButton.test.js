import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import ActionSearch from '@material-ui/icons/Search';

import AddParticipantDialogIconButton from '../AddParticipantDialogIconButton';

configure({ adapter: new Adapter() });

describe('<AddParticipantDialogIconButton />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const children = (<ActionSearch />);

      // Act
      const renderedComponent = shallow(<AddParticipantDialogIconButton>{children}</AddParticipantDialogIconButton>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = (<ActionSearch />);

      // Act
      const renderedComponent = shallow(<AddParticipantDialogIconButton>{children}</AddParticipantDialogIconButton>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('style tests', () => {
    it('should have material-ui props for styling', () => {
      // Arrange
      const children = (<ActionSearch />);
      const expectedProps = {
        style: { top: '26px', height: '30px' },
      };

      // Act
      const renderedComponent = shallow(<AddParticipantDialogIconButton>{children}</AddParticipantDialogIconButton>);

      // Assert
      expect(renderedComponent.props().contentStyle).toEqual(expectedProps.contentStyle);
    });
  });
});
