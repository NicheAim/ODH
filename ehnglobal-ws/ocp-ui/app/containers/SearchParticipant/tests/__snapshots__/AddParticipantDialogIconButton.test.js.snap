// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddParticipantDialogIconButton /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(IconButton)>
    <pure(Search) />
  </Styled(IconButton)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <pure(Search) />,
      "className": "sc-bdVaJa bDWFJH",
      "disableTouchRipple": false,
      "disabled": false,
      "iconStyle": Object {},
      "style": Object {
        "height": "30px",
        "top": "26px",
      },
      "tooltipPosition": "bottom-center",
      "touch": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <pure(Search) />,
        "className": "sc-bdVaJa bDWFJH",
        "disableTouchRipple": false,
        "disabled": false,
        "iconStyle": Object {},
        "style": Object {
          "height": "30px",
          "top": "26px",
        },
        "tooltipPosition": "bottom-center",
        "touch": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
